Core/Store.swift

```diff
@@ -148,10 +148,11 @@ public class Store<Product: Purchaseable> {
                                                                 self.dismissAvailablePurchasesModal(wasCancelled: true)
         })
         
-        presentingViewController.present(modalViewController, animated: true, completion: nil)
         self.modalViewController = modalViewController
-        refreshProductsList()
         
+        presentingViewController.present(modalViewController, animated: true) { [weak self] in
+            self?.refreshProductsList()
```

Comment:
This will not call `refreshProductsList()` until the presentation has completely finished its animation(s). This will certainly alleviate the nil outlet issue; but does calling it later affect the timing at all? I'd mostly worry about un-sightly content "pop-in" as you see an empty screen animate in with the presentation and only fill up when it fully appears.
If there are visual or timing artifacts, then this looks like a good approach.

Reply:
That is a good question. You are right that calling the refresh after the view is loaded and animations are finished will delay the loading of the products. I don't think that will be a major UX hit here because 1) that time is relatively short and 2) the `IAPDialogViewController` loads with an activity indicator animating by default. The activity indicator will gracefully animate away once the products are loaded.

Does that address your concern?

Reply:
Absolutely! I just wanted to check because I couldn't run that particular use case.